tests/saveload_macro.rs

```diff
@@ -0,0 +1,147 @@
+#![cfg(feature = "serde")]
+
+#[macro_use]
+extern crate specs;
+#[macro_use]
+extern crate specs_derive;
+#[macro_use]
+extern crate shred_derive;
+#[macro_use]
+extern crate serde;
+extern crate ron;
+extern crate serde_json;
+extern crate shred;
+
+use specs::prelude::*;
+use specs::saveload::{U64Marker, U64MarkerAllocator};
+
+#[derive(Copy, Clone, Serialize, Deserialize, Component, Debug, PartialEq)]
+#[storage(VecStorage)]
+pub struct Pos {
+    x: f64,
+    y: f64,
+}
+
+#[derive(Copy, Clone, Serialize, Deserialize, Component, Debug, PartialEq)]
+#[storage(VecStorage)]
+pub struct Vel {
+    x: f64,
+    y: f64,
+}
+
+#[derive(Copy, Clone, Component, Saveload)]
+#[storage(VecStorage)]
+pub struct OwnsEntity(Entity);
+
+saveload_components!{
+    [Pos, Vel, OwnsEntity], DeData, SerData, Data
+}
+
+#[derive(Clone, Eq, PartialEq, Hash, Default, Debug)]
+struct Serialized {
+    json: String,
+}
+
+#[derive(SystemData)]
+struct DeSysData<'a> {
+    de: saveload_generated::DeData<'a, U64Marker, U64MarkerAllocator>,
```

Comment:
We had a `WorldSerialize` before that had this convenience without macros on the user side.